### PR TITLE
adds package dependencies based on pipenv and pip tools

### PR DESCRIPTION
**Pull request by sebastienraillet**
# Summary
As stated inside #12, this repository is lacking dependency files to easily install through _pipenv_ or _pip_ tools the dependencies needed by vsc_tools

# Description
* adds Pipfile and Pipfile.lock to allow to easily install through pipenv the needed dependencies used by vsc tools
* adds requirements.txt to allow the same above usage but with pip install
* README.md is updated in accordance

# Notes
- This work is mainly inspired by what has been done inside vss-tools
- I haven't do the job about setuptools files. Let me know if this part is needed or not ? Knowing that currently, seems there are still a lot of work in progress inside this repository (let me know if I'm wrong)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,8 +7,58 @@ Primarily, it allows reading a service description and generating various
 types of output.  It uses the [Jinja2 templating language](https://jinja.palletsprojects.com) ([(alt. link)](https://jinja2docs.readthedocs.io/en/stable/))
 for most output definitions.
 
-# Testing it out
+## Getting started
 
+### Prerequisites
+* Python 3.7 installed
```
Comment (gunnarx) [inline README.md:13]: 3.8

**Review by gunnarx — CHANGES_REQUESTED** [review]

**Comment by gunnarx** [comment]
Thanks for this.  Sorry for the slow review response.

File: README.md (hunk 1 above)
Comment (sebastienraillet) [reply README.md]: Right, I've made a force push on the branch to solve this. Should be ok now

**Comment by gunnarx** [comment]
> I haven't do the job about setuptools files. Let me know if this part is needed or not

I'll accept whatever you think here, what is "best practice".

> Knowing that currently, seems there are still a lot of work in progress inside this repository (let me know if I'm wrong)

I agree, it is too early to start releasing versioned packages on various external sites.  But at any time we can of course make the necessary preparations for it.

**Review by gunnarx — APPROVED** [review]

**Comment by gunnarx** [comment]
Looks good to me.  Anyone else want to take a look @erikbosch @magnusfeuer ?

File: README.md [hunk 2]
```diff
@@ -7,8 +7,58 @@ Primarily, it allows reading a service description and generating various
...
+[pipenv](https://pypi.org/project/pipenv/) is a tool that manages a virtual environment and install the package and its dependencies, making the process much simpler and predictable, since the `Pipfile` states the dependencies, while `Pipfile.lock` freezes the exact version in use.
+
+If [`pyenv` shell command](https://github.com/pyenv/pyenv) is not installed, use its [installer](https://github.com/pyenv/pyenv-installer) to get it:
+
+```sh
+curl https://pyenv.run | bash  # download and install
+exec $SHELL                    # restart your shell using the new $PATH
+```
+
+Make sure Python version 3.8.5 is installed:
```
Comment (erikbosch) [inline README.md:46]: Does it necessarily have to be 3.8.5, or would any 3.8.X-version do? If it is not strict we could add something like `(or any later 3.8.X version, if available)`.

I am no Pip expert , but the Pipfile says ` python_version = "3.8"` so I assume that any 3.8.X version would suffice, or?

Comment (sebastienraillet) [reply README.md:46]: Hi @erikbosch,

To be honest about python version, I've take the same version that one used on vss-tools and made the test with that one.
But you're right, there is an inconsistency between Pipfile and documentation.

By having a look to python 3.8 series changelog [here](https://docs.python.org/3.8/whatsnew/changelog.html#python-3-8-12), it seems changes between each releases is only about bugfix or security issues.

I've also made a test with python 3.8.0 and python 3.8.12. Both are working properly (I haven't tried each versions between 😉). Moreover, when you are using _pyenv_ and _pipenv_ together and that Pipfile mentions only 3.8, _pyenv_ will automatically download the latest python version available for this minor version (for 3.8, it is 3.8.12 actually).

If everyone agree, I can adapt the documentation to indicate that python version of 3.8.x series will do the job in any cases.

**Comment by erikbosch** [comment]
@gunnarx - is this one still something that shall be merged?

**Comment by gunnarx** [comment]
Looking at it.  Rebased on latest master above.

**Pull request merged**